### ISSUE TITLE
Preload stylesheet tags differ from vite preload stylesheet causing CORS issue from cache

### ISSUE THREAD

**Issue by smdern** [ElMassimo/vite_ruby#189]
### Description 📖

`vite_javascript_tag` injects stylesheet tags using `crossorigin: anonymous` [see here](https://github.com/ElMassimo/vite_ruby/blob/851d617d7525168d3217dfbe4055deae2c21afff/vite_rails/lib/vite_rails/tag_helpers.rb#L37).  Setting `crossorigin` on the stylesheet tag is fine, except that it conflicts with how vite's [preload-helper injects stylesheet link tags](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/importAnalysisBuild.ts#L67-L70) where they don't a `crossOrigin` attribute on the link tag for css resources.

Example: 
```html
<!--Vite-Rails renders-->
<link rel="stylesheet" crossorigin="anonymous" href="//example.com/application.f372e162.css" media="screen">
<!--Vite.js preload-helper injects-->
<link rel="stylesheet" href="//example.com/application.f372e162.css">
```

This can cause an issue with browser caching, where the first link tag will request the stylesheet with with an `Origin` header set (because of the crossorigin attribute is set), but then the second request (the link injected from vite's preload-helper) will request the the same resource but since `crossorigin` isn't present on the `<link>` tag, the origin header isn't present, causing the browser to cache the response without `access-control-allow-origin` header present. 

When the page is refreshed the `link` tag that had `crossorigin="anonymous"` set will not succeed due to the cached response not including `Access-Control-Allow-Origin`. 

This can be solved from the server by ensuring  `Vary: Origin` is set in the header, but that will actually cause the same resource to be fetched twice and having two separate cache entries.

We were able to work around this issue by using the `skip_style_tags: true` option with `vite_javascript_tag` and then using a `stylesheet_link_tag(*ViteRuby.instance.manifest.resolve_entries("application", type: :javascript)[:stylesheets])`. By not including a `crossorigin` attribute we are now injecting a `link` tag that is compatible with vite preload-helper.

I opened the issue here since `vite-ruby` depends on `vite`. This project might be doing the correct thing, but either way there is an inconsistency.

Thank you

**Comment by ElMassimo** [comment]
@smdern Thanks for reporting!

Vite Ruby should work well with Vite's defaults, so it makes sense to [avoid adding the `crossorigin` attribute to the stylesheets](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/lib/vite_rails/tag_helpers.rb#L37).

Let me know if you would be interested in making a PR.

**Comment by smdern** [comment]
Yeah, I can put together a PR. Probably won't happen until early nest week. If you're in a position to make the change sooner, I won't be offended ;)

Thank you for the quick response.

**Comment by smdern** [comment]
https://github.com/ElMassimo/vite_ruby/pull/190 should resolve this issue.

**Comment by ElMassimo** [comment]
Thanks again! Released in [`vite_rails@3.0.6`](https://github.com/ElMassimo/vite_ruby/blob/d18ddaa7a14e84ce03ee2d83a5eb982aa8234e56/vite_rails/CHANGELOG.md#306-2022-02-04).

**Comment by smdern** [comment]
No problem. Thank you for the heads-up with the release.
